### PR TITLE
Implement PlatformLocale and StringDelegate for k/wasm (#561)

### PR DESCRIPTION
**Pull request by eymar**
This PR "upstreams"  https://github.com/JetBrains/compose-multiplatform-core/pull/561 from wasm-main to jb-main

### CODE REVIEW

File: compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt [hunk 1]
```diff
@@ -16,28 +16,54 @@
...
 
         override fun parseLanguageTag(languageTag: String): PlatformLocale {
-            return JsLocale(Any())
+            return JsLocale(languageTag)
         }
     }
 
-internal actual fun PlatformLocale.isRtl(): Boolean = false // TODO
+
+private val rtlLanguagesSet = setOf("ar", "fa", "he", "iw", "ji", "ur", "yi")
```
Comment (igordmn) [inline compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt:51]: What is the source of this? Can we add a link in a comment?

[This list](https://meta.wikimedia.org/wiki/Template:List_of_language_names_ordered_by_code) has more RTL languages

**Review by igordmn — APPROVED** [review]

**Review by MatkovIvan — APPROVED** [review]

File: compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt:51]: As far as I remember it's from sources of function that we call on java target

Comment (eymar) [inline compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt]:
```suggestion
// The list of RTL languages is taken from https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/java/awt/ComponentOrientation.java#L156
private val rtlLanguagesSet = setOf("ar", "fa", "he", "iw", "ji", "ur", "yi")
```

Comment (eymar) [reply compose/ui/ui-text/src/jsWasmMain/kotlin/androidx/compose/ui/text/intl/JsPlatformLocale.js.kt:51]: That's right. I added a comment.

It was taken from here https://github.com/openjdk/jdk/blob/master/src/java.desktop/share/classes/java/awt/ComponentOrientation.java#L156 to follow the desktop impl.

**Pull request merged**